data/block/blockV2.go

```diff
@@ -507,7 +569,26 @@ func (hv2 *HeaderV2) SetAdditionalData(headerVersionData headerVersionData.Heade
 	if check.IfNil(headerVersionData) {
 		return data.ErrNilPointerDereference
 	}
-	return hv2.SetScheduledRootHash(headerVersionData.GetScheduledRootHash())
+
+	err := hv2.SetScheduledRootHash(headerVersionData.GetScheduledRootHash())
+	if err != nil {
+		return err
+	}
+
+	if headerVersionData.GetScheduledRootHash() == nil {
```

Comment:
could be moved before the previous assignment?

`err := hv2.SetScheduledRootHash(headerVersionData.GetScheduledRootHash())`

Reply:
I think now this can be completely removed.
In this case additional data values will be zero values for all fields.

Reply:
ok